From 29beb53ab085507b466d9acac7555c3f03caacbd Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Thu, 5 Dec 2024 17:44:59 +0800
Subject: [PATCH 089/115] mac80211: mtk: add support QoS prgram related bit in
 extended capabilities

Add support QoS Map information bit (bit 32) and Mirrored SCS bit (bit
85) in extended capabilities field.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 include/linux/ieee80211.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/linux/ieee80211.h b/include/linux/ieee80211.h
index ae63ec61..e1f4103c 100644
--- a/include/linux/ieee80211.h
+++ b/include/linux/ieee80211.h
@@ -4133,6 +4133,9 @@ enum ieee80211_tdls_actioncode {
  */
 #define WLAN_EXT_CAPA4_INTERWORKING_ENABLED	BIT(7)
 
+/* QoS Map information support */
+#define WLAN_EXT_CAPA5_QOS_MAP	BIT(0)
+
 /*
  * TDLS capabililites to be enabled in the 5th byte of the
  * @WLAN_EID_EXT_CAPABILITY information element
@@ -4173,6 +4176,9 @@ enum ieee80211_tdls_actioncode {
 /* Enable Beacon Protection */
 #define WLAN_EXT_CAPA11_BCN_PROTECT	BIT(4)
 
+/* Mirrored stream classification service support */
+#define WLAN_EXT_CAPA11_MIRRORED_SCS_SUPPORT	BIT(5)
+
 /* TDLS specific payload type in the LLC/SNAP header */
 #define WLAN_TDLS_SNAP_RFTYPE	0x2
 
-- 
2.45.2

